sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -363,6 +363,38 @@ pi_result _pi_device::initialize() {
   ZeDeviceComputeProperties.version =
       ZE_DEVICE_COMPUTE_PROPERTIES_VERSION_CURRENT;
   ZE_CALL(zeDeviceGetComputeProperties(ZeDevice, &ZeDeviceComputeProperties));
+
```

Comment:
What's the motivation of this? The tests that don't need many (any) of these properties will see increased total time as this caching is now unconditionally performed at device initialization. A better approach could be still doing these in piDeviceGetInfo, but only once for PI device.

Reply:
When you see the traces for a typical application, you see that all piDeviceGetInfo is called a lot, and everytime you are doing the get call to Level-Zero, so we could use a way of minimizing that overhead (those properties are not gonna change from call to call).

One option is this, other option would be adding it to deviceGetInfo(), having a flag to indicate they were already retrieved. I was trying to avoid that, since we would to control concurrent access. Basically, we would be trading off overhead of getting the properties at initialization, vs overhead of acquiring lock for the new flag. We could get away with not having the flag, since all that we are doing here is reading properties, but still, we are writing to pi_device's fields. Let me know what you think.

Reply:
I think we should now continue with what you are doing here, but add a note comment that we should consider the in-piDeviceGetInfo caching if hitting too large start-up times